backend/Dockerfile

```
@@ -0,0 +1,30 @@
+# Stage 1
+# dependencies
+FROM node:16 as dependencies
+WORKDIR /app
+COPY package*.json ./
+RUN npm ci
+
+# Stage 2
+# prune 
+FROM dependencies as prune
+RUN npm prune --production
+
+# Stage 3
+# build
+FROM dependencies as build
+COPY tsconfig.json ./
+COPY src ./src
+RUN npm run build
+
+# Stage 4
+# release
+FROM dependencies 
```

```suggestion
FROM node:16
```

Or else the whole non-pruned `node_modules` will be in container layers